### ISSUE TITLE
Use what software on Windows to compile for ESP8266

### ISSUE THREAD

**Issue by Fireflaker** [cnlohr/colorchord#52]
Sorry if it is a stupid question. How should I compile the bin file for ESP8266? Now I have the MSYS setup but I do not have the exe file which is needed in "mingw32-make colorchord.exe" 

Thank you

**Comment by cnlohr** [comment]
Hmm, I was trying to get it to work with TCC in windows.  I'll try it again and report back.

**Comment by Fireflaker** [comment]
Thank you! I will try it out.

CNLohr <notifications@github.com>于2017年9月10日 周日下午8:27写道：

**Comment by Fireflaker** [comment]
It seems like TCC is no longer in development from 2013. Are there other
compilers that can compile all these c files into bin for esp8266? I would
prefer one ending in exe because I am terrible in proficiency.

Would it be possible to upload a compiled bin file for the esp8266 so I can
just burn it?
Or should I install ubuntu to compile it.

Thanks...

wzd leo <leowzd@gmail.com>于2017年9月10日 周日下午8:41写道：

**Comment by cnlohr** [comment]
TCC is funny.  It's in VERY active development.  Literally daily commits.  Just no one's made a release.

I think I have it almost 100% windows compatible.  Please check it out again, then look in colorchord2/windows/compile.bat

You will need to install TCC 0.9.26 to C:\tcc to use it.  

On my Windows system it seems not to get any sound data from the sound callback.  Could you check?

Re: ESP8266.  I will consider that.  I have a few tweaks I want to bring into the trunk.  Will begin shortly.

To compile for the ESP8266, it is much easier to use Ubuntu.

**Comment by Fireflaker** [comment]
Got it. I will check out. Thank you for your help.

CNLohr <notifications@github.com>于2017年9月10日 周日下午11:04写道：

**Comment by Fireflaker** [comment]
I made the exe file using TCC and it works OK on my win10 laptop. It picks
up sound from the laptop's internal microphone but I can not make it
monitor the audio output. Another minor bug is the FPS says 0 at all time.
Maybe check if you have a microphone on your windows device or if you have
it disabled.

How can I get the binary file for the embedded ESP8266 ColorChord?

Thank you

2017-09-11 7:19 GMT-04:00 wzd leo <leowzd@gmail.com>:

**Comment by cnlohr** [comment]
That's really good that it works.  Really bad that it doesn't work on my Windows installation.  I spent most of my time trying to figure that out :(.

Windows 10 does not support monitor as a default devices anymore.  You'll have to install Cable. http://www.vb-audio.com/Cable/#DownloadCable

There's a flag like WinAudioNumber or something like that in the default.ini.

The binaries are sitting in the main folder for esp are sitting in the root embedded8266 folder.  If you have an old board, you may need to also flash it's init.bin files.  I don't know how to do that from Windows though :-/

Closing issue. You may continue to comment, though.

**Comment by Fireflaker** [comment]
Thank you.
I have already uploaded these bin files but as I connect to the ESP8266 and
go to 192.168.4.1, it says File not found. Should I upload some other files
in the same folder? How should I do it? The only port opened is No.80 for
http. Sorry as these questions may be stupid.

CNLohr <notifications@github.com>于2017年9月11日 周一下午9:42写道：

**Comment by Fireflaker** [comment]
This is what I got in the sirial monitor as I try to go to 192.168.4.1

esp82XX Web-GUI
Version: v0.5-26-gfaf53-dev - Build Mon, May 22 2017, 21:56:04 -0400 with -DICACHE_FLASH -DSK6812 -DWEB_PORT=80 -DCOM_PORT=7777 -DBACKEND_PORT=7878  -DSLOWTICK_MS=50Loading Settings: af / 0 / 69 / 69
Settings Loaded: ESP_17B8D2 / Default
Opmode: 2
Default SoftAP mode: "ESP_17B8D2":""
Boot Ok.
IGMP Joining: 0104a8c0 fb0000e0
MFS Not found at regular address (ffffffff).
MFS Not found at alternative address (ffffffffffffffff).
MFS Found at: 00000000
404(index.html)
MFS Not found at regular address (ffffffff).
MFS Not found at alternative address (ffffffffffffffff).
MFS Found at: 00000000
404(favicon.ico)

**Comment by cnlohr** [comment]
I forgot you will need to flash the web binary.

You may either flash it to exactly 0x10000 (65536) or 0x100000 (1MB boundary). https://github.com/cnlohr/colorchord/blob/master/embedded8266/web/page.mpfs

**Comment by Fireflaker** [comment]
Thanks! Got it working! I would not get any of these to work without your
help! 😙


CNLohr <notifications@github.com>于2017年9月11日 周一下午11:18写道：

**Comment by cnlohr** [comment]
If you can experiment with the Windows build and see if you can get it working with CABLE.  Also, if you had any problems please let me know since I still can't get it to work on my Windows install.

**Comment by Fireflaker** [comment]
Thanks for your help. 
What do you mean by CABLE? I do not have linux installed and I am pretty terrable at coding. I tried to change the audio input source but got no response. I also tried an USB sound card and set it as defult but got no sound. It seems to work only with the built in microphone.

I do have another problem with embedded ESP8266 but I will ask under another issue.

**Comment by cnlohr** [comment]
http://www.vb-audio.com/Cable/#DownloadCable ... and you should be able to select default sound device in Windows to pick which one.  But that's an interesting observation.  Maybe that feature isn't working right for me, and I'm only able to capture from a null device.

**Comment by Fireflaker** [comment]
![Uploading IMG_2714.JPG…]()
![Uploading IMG_2720.JPG…]()
Anyways, the ESP8266 is working perfectly. THANK YOU!

**Comment by hatf0** [comment]
@cnlohr I was able to get Cable working with ColorChord on Windows 10. 
https://hastebin.com/icehafarun.config
This was my config-- and I ended up setting the VAC output to the default, then setting a loopback on the recording device to output to my speakers. Hacky, but it works. 
https://cdn.discordapp.com/attachments/262128321314291712/422647281720754176/2018-03-12_00-49-00.mkv demonstration- proof is in the pudding!
